components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/UserNameMapperImpl.java

```
@@ -0,0 +1,113 @@
+     */
+    @Override
+    public String getLoggedInUserIDFromPseudoName(String pseudoName) {
+        //TODO implement method
+        //If pseudo name in map then get from that
+        //Else check mapping in database and load it to local map.
+        //If mapping is not in database then add it to db and cache both.
+        //then return name.
+        if (pseudoName != null && pseudoName.equalsIgnoreCase("admin")) {
+            return pseudoName;
+        } else {
+            try {
+                String userID = getUserMappingDAO().getUserIDByPseudoName(pseudoName);
+                return userID;
+            } catch (APIMgtDAOException e) {
+                log.error("Error while user getting user details for user : " + pseudoName);
+                return null;
+            }
+        }
+    }
+
+    /**
+     * This method will return logged in user's pseudo name(pseudo name used within APIM domain) using provided
+     * user id.
+     *
+     * @param userID user identifier of the user who need pseudo name.
+     * @return pseudo name of the matching user.
+     */
+    @Override
+    public String getLoggedInPseudoNameFromUserID(String userID) {
+        //TODO implement method
+        //If userName in map then get from that
+        //Else check mapping in database and load it to local map.
+        //then return name.
+        if (userID != null && userID.equalsIgnoreCase("admin")) {
+            return userID;
+        } else {
+            try {
+                String pseudoName = getUserMappingDAO().getPseudoNameByUserID(userID);
+                return pseudoName;
+                //return APIManagerFactory.getInstance().getIdentityProvider().getIdOfUser(userName);
+            } catch (APIMgtDAOException e) {
+                //Should not log real user identity due to any reason.
+                log.error("Error while user getting user details for user : XXX");
+                return null;
```

Since this a clear error condition, lets throw an exception here instead of returning null and risking a NPE